src/app/dataset-view/additional-components/metadata-component/components/add-push-source/add-push-source.component.ts

```diff
@@ -76,11 +71,62 @@ export class AddPushSourceComponent extends BaseMainEventComponent {
         }),
     });
 
-    public get readForm(): FormGroup {
+    public initEditForm(): void {
+        this.trackSubscription(
+            this.editService
+                .getEventAsYaml(this.getDatasetInfoFromUrl(), this.queryParamName ?? "")
+                .subscribe((result) => {
+                    this.history = this.editService.history;
+                    if (this.unsupportedSourceName()) {
+                        this.zone.run(() => {
+                            this.navigationServices.navigateToPageNotFound();
+                        });
+                    }
+                    if (result) {
+                        this.eventYamlByHash = result;
+                        const currentPushSourceEvent = this.editService.parseEventFromYaml(this.eventYamlByHash);
+                        console.log("==>", currentPushSourceEvent);
+                        this.addPushSourceForm.patchValue({ sourceName: currentPushSourceEvent.sourceName });
+                    }
+                    if (!this.queryParamName) {
+                        this.addPushSourceForm.controls.sourceName.addValidators(
+                            RxwebValidators.noneOf({
+                                matchValues: [...this.getAllSourceNames(this.history), "default"],
```

Comment:
There will be no `default` in the ODF, as we agreed wtih SM. There will be mandatory unique names.

Reply:
Fixed